sys/Makefile.dep

```diff
@@ -2,6 +2,7 @@ ifneq (,$(filter arduino,$(USEMODULE)))
   FEATURES_OPTIONAL += periph_i2c
   FEATURES_OPTIONAL += periph_spi
   FEATURES_REQUIRED += periph_uart
+  FEATURES_REQUIRED += cpp
```

Comment:
atmega CPUs doesn't provide the cpp feature so this will prevent arduino boards to run arduino applications although some of them provides the arduino feature.

Reply:
Oha! (But the feature requirement is still correct. I will check tonight if I can swiftly add `cpp` to the ATmega platform, so that this PR does not cause backlash from AVR/Arduino users.)

Reply:
There are non-C++ Arduino applications oO?

Reply:
IIRC, the problem is the AVR toolchain that doesn't fully support c++.

Reply:
(already spent some time on that issue)

Reply:
Well technically, the language used with the Arduino IDE is [Processing](https://en.wikipedia.org/wiki/Processing_(programming_language)) ;-P

Reply:
Isn't Processing the language the IDE os written in? But I also recall having read that the Arduino guys refer to their C++ with some custim libraries as a new language.

I think that the AVR toolchain does support C++. However, the Ubuntu package is sadly horribly outdated and has been unmaintained for years. I remember having written some code using `sys/arduino` that worked fine for me, but didn't work for a student due to the Ubuntu toolchain. Maybe we can just build the toolchain from source in the `Dockerfile`and ask Ubuntu users to just build with `BUILD_IN_DOCKER=1`?

Reply:
OK, it is pretty straight forward to allow *some* C++ on AVR. But there is no libstdc++ for AVR, so most C++ code just won't work.

How about we split the `cpp` feature into two features, with the following meaning

- `cpp`: Basic C++ language support, including statically allocated class instances. (I have [a branch](https://github.com/maribu/RIOT/tree/avr-cpp) that would provide this.)
- `libstdcpp`: Proper C++ support, including a working `libstdc++`

The Arduino stuff only needs basic C++ support, so it could depend on `cpp`. "Real" C++ applications needing a proper `libstdc++` could depend on `libstdcpp` to avoid having to blacklist all AVR boards for them.

Reply:
Or maybe `cpp-basic` and `cpp`? Or `cpp`basic` and `cpp-full`?

Reply:
I think just adding an extra feature called `libstdcpp` would be clearer to the end user on what is actually used in their builds. This also could add basic c++ support when you don't want all the features of c++ but only header only stuff.

Reply:
Should be solved by https://github.com/RIOT-OS/RIOT/pull/14598